hypershift-operator/controllers/hostedcluster/internal/platform/agent/agent.go

```diff
@@ -173,6 +173,55 @@ func (p Agent) ReconcileCredentials(ctx context.Context, c client.Client, create
 	if err != nil {
 		return fmt.Errorf("failed to reconcile Agent RoleBinding: %w", err)
 	}
+
+	return p.reconcileClusterRole(ctx, c, createOrUpdate, controlPlaneNamespace)
+}
+
+func (p Agent) reconcileClusterRole(ctx context.Context, c client.Client, createOrUpdate upsert.CreateOrUpdateFN,
+	controlPlaneNamespace string) error {
+
+	role := &rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: credentialsRBACName,
+		},
+	}
+	_, err := createOrUpdate(ctx, c, role, func() error {
+		role.Rules = []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{"cluster.open-cluster-management.io"},
+				Resources: []string{"managedclustersets/join"},
```

Comment:
You may have said this already but are `managedclustersets` cluster-scoped?

Reply:
yep, very pity but yes